src/scenes/TaskDetailsView/index.tsx

```diff
@@ -1,73 +1,73 @@
 import {Profile} from "../../types/Profile";
-import {useContext} from "react";
+import {useContext, useEffect, useState} from "react";
 import userContext from "../../types/UserContext";
 import {Badge, Button, Card} from "flowbite-react";
 import {HiCheck, HiPencil, HiTrash} from "react-icons/hi";
+import {Task} from "../../types/Task";
+import {TaskService} from "../../services/taskService";
+import {useParams} from "react-router-dom";
 
 
 interface TaskDetailsViewProps {
     isAgentView: boolean,
   }
 
 function TaskDetailsView({ isAgentView }: TaskDetailsViewProps){
+  const profile = useContext(userContext);
+  const { id } = useParams();
+  const [task ,setTask] = useState<Task | null>(null);
 
-    const profile = useContext(userContext);
+  useEffect(() => {
+    fetchTask();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-    return(
-      
+  const fetchTask = async () => {
+    setTask(await TaskService.getTaskById(id!));
+  }
+
+  return(
     <div className="px-5">
       <Card>
         <div className="px-5 space-y-8">
-
           <div className="space-y-2">
-            <h1 className="text-3xl font-medium text-gray-900">Task Name</h1>
+            <h1 className="text-3xl font-medium text-gray-900">{task?.title}</h1>
             <div className="flex flex-wrap items-center gap-2">
-                <Badge color="info">Developer Task</Badge>
-
-                {!isAgentView && (
-                  <Badge color="success" icon={HiCheck}>
-                    <div className="px-1">
-                      Completed
-                    </div>
-                  </Badge>
-                )}
+              <Badge color="info">Developer Task</Badge>
```

Comment:
check `task?.type` and conditionally render the badge either developer task or quiz task

Reply:
Fixed with 19c6e84. Please review. @Gravewalker666